File: solutions/elixir/065-maximum-depth-of-binary-tree.ex
# Problem 65: Maximum Depth of Binary Tree (LeetCode 104)
# Difficulty: Easy
# Language: Elixir
# 
defmodule MaxDepth do
  def max_depth(nil), do: 0
  def max_depth({_, l, r}), do: 1 + max(max_depth(l), max_depth(r))
end

File: solutions/typescript/065-maximum-depth-of-binary-tree.ts
/**
 * Problem 65: Maximum Depth of Binary Tree (LeetCode 104)
 * Difficulty: Easy
 * Language: TypeScript
 */
type TreeNode = { val: number; left: TreeNode | null; right: TreeNode | null };

function maxDepth(root: TreeNode | null): number {
  return root === null ? 0 : 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
}
